### PR TITLE
Separate free and paid domains in domain picker and fix empty state on mobile

### PR DESCRIPTION
**Pull request by alshakero**
#### Changes proposed in this Pull Request

1. It separates domains into two subgroups, free and paid.
2. It fixes the empty state on mobile. Currently, it looks like this:
<kbd></kbd>
After the change, it looks like this:
<kbd></kbd>

#### Testing instructions

1. Search for a domain, placeholders should have the same structure as results.
2. Search for a very long domain (50+ chars), the API will give only free option, results should look sane (the Custom domains subgroup shouldn't show up at all).
3. Test on mobile. It should look as you'd expect.
4. Test empty state by emptying the search field. On Desktop and mobile (Chrome DevTools is more than enough).

Fixes https://github.com/Automattic/wp-calypso/issues/43823

#### For translators:
Screenshot for translators:
image

### CODE REVIEW

File: packages/domain-picker/src/domain-picker/suggestion-item.tsx [hunk 1]
```diff
@@ -82,7 +82,7 @@ const DomainPickerSuggestionItem: FunctionComponent< Props > = ( {
 				} ) }
 			>
 				{ suggestion.is_free ? (
-					__( 'Free' )
+					__( 'Included in free plan' )
```
Comment (simison) [inline packages/domain-picker/src/domain-picker/suggestion-item.tsx:85]: i18n bot will come and ask for screenshots for this change. ;-)

Comment (alshakero) [reply packages/domain-picker/src/domain-picker/suggestion-item.tsx:85]: Attached

File: packages/domain-picker/src/domain-picker/index.tsx [hunk 2]
```diff
@@ -170,29 +160,67 @@ const DomainPicker: FunctionComponent< Props > = ( {
...
+										<Button onClick={ () => setIsExpanded( true ) }>
+											{ __( 'View more results' ) }
+										</Button>
+									</div>
+								) }
+						</div>
+					) : (
+						<div className="domain-picker__suggestion-sections">
+							<div className="domain-picker__suggestion-item-group">
+								<p className="domain-picker__suggestion-group-label">{ __( 'Sub-domain' ) }</p>
```
Comment (razvanpapadopol) [inline packages/domain-picker/src/domain-picker/index.tsx:213]: This duplication of markup could be avoided if we'd replace the `&&` with a ternary at https://github.com/Automattic/wp-calypso/pull/43830/files#diff-f948bcd10a91168abd35256bd0ae22aaR167 and render the placeholder there.

File: packages/domain-picker/src/domain-picker/index.tsx [hunk 3]
```diff
@@ -170,29 +160,67 @@ const DomainPicker: FunctionComponent< Props > = ( {
...
-							/>
-						) ) ?? times( quantity, ( i ) => <SuggestionItemPlaceholder key={ i } /> ) }
-						{ ! isExpanded &&
-							allDomainSuggestions?.length &&
-							allDomainSuggestions?.length > quantity && (
-								<div className="domain-picker__show-more">
-									<Button onClick={ () => setIsExpanded( true ) }>
-										{ __( 'View more results' ) }
-									</Button>
+					{ domainSuggestions ? (
```
Comment (razvanpapadopol) [inline packages/domain-picker/src/domain-picker/index.tsx:163]: This ternary wouldn't be necessary if we render the placeholders exactly where the items are rendered when `domainSuggestions` are loaded.

**Pull request merged**